### ISSUE TITLE
[task_executor] iteration over non-sequence

### ISSUE THREAD

**Issue by cdondrup** [strands-project/strands_executive#204]
Apparently, when the scheduler has to preempt a task before the robot reached the destination and started execution, the task executor dies:

```
[INFO] [WallTime: 1431359349.086323] No schedule found
[INFO] [WallTime: 1431359349.090286] Pausing execution
[INFO] [WallTime: 1431359349.091270] Interrupting active task
[INFO] [WallTime: 1431359349.091519] Requesting preempt on state machine in state ['TASK_NAVIGATION']
[INFO] [WallTime: 1431359349.091745] Waiting for exit
[INFO] [WallTime: 1431359349.092082] Preempt requested on action 'mdp_plan_exec/execute_policy'
[INFO] [WallTime: 1431359349.092471] Preempt on action 'mdp_plan_exec/execute_policy' cancelling goal: 
task_type: 0
target_id: WayPoint12
[INFO] [WallTime: 1431359349.351872] Topological navigation execute policy action server exited with status: PREEMPTED
[INFO] [WallTime: 1431359349.361936] Concurrent state 'MONITORED' returned outcome 'preempted' on termination.
[INFO] [WallTime: 1431359349.687005] Concurrent state 'MONITORING' returned outcome 'preempted' on termination.
[WARN] [WallTime: 1431359349.693454] State 'MONITORING' in concurrence did not service preempt.
[INFO] [WallTime: 1431359349.693730] Concurrent Outcomes: {'MONITORED': 'preempted', 'MONITORING': 'preempted'}
[INFO] [WallTime: 1431359349.693898] Nav terminated with outcome preempted
[INFO] [WallTime: 1431359349.696123] State machine transitioning 'TASK_NAVIGATION':'preempted'-->'TASK_CANCELLED'
[INFO] [WallTime: 1431359349.696575] Preempt requested on state machine before executing the next state.
[INFO] [WallTime: 1431359349.696719] Last state 'TASK_NAVIGATION' did not service preempt. Preempting next state 'TASK_CANCELLED' before executing...
[INFO] [WallTime: 1431359349.696882] Execution of task 2 was cancelled
[INFO] [WallTime: 1431359349.698905] State machine terminating 'TASK_CANCELLED':'preempted':'preempted'
[INFO] [WallTime: 1431359349.706554] And relax
Exception in thread Thread-11:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/cdondrup/ros_ws/src/strands_executive/task_executor/scripts/scheduled_task_executor.py", line 669, in schedule_tasks
    self.try_schedule(unscheduled)
  File "/home/cdondrup/ros_ws/src/strands_executive/task_executor/scripts/scheduled_task_executor.py", line 522, in try_schedule
    for taskU in self.unscheduled_tasks:
TypeError: iteration over non-sequence
```

Clumsy fingers sent this before it was completed...

Anyidea? maybe @hawesie or @bfalacerda ?

**Comment by cdondrup** [comment]
Mh, apparently it's also happening when the action is already executing. So as soon as it has to cancel a task that is executing it dies.

**Comment by mudrole1** [comment]
Yes, that is mine :( almost there!

**Comment by mudrole1** [comment]
Ok, this should be now fixed, sorry! See PR #200
